submissions/angels_unit_tests/angels_unit_tests.md

```diff
@@ -0,0 +1,620 @@
+
+![Settings within Angel's Industries](https://raw.githubusercontent.com/LovelySanta/Alt-F4/angels-unit-tests/submissions/angels_unit_tests/angels_industry_settings.png)
+
+Since the loading and saving part of mod settings is completely handled by the base game, it was a part that I didn't encounter in creating mods before. However, I was very surprised at finding [thorough documentation on the Factorio Wiki](https://wiki.factorio.com/Mod_settings_file_format) about this.
+
+[The second script](https://github.com/Arch666Angel/mods/pull/634/commits/abb64c2fe8007e060f112716ebe33864510b6451) configures the `mod-list.json` file, which contains the information about which mods have to be enabled. This is the part which are people most familiar with that play modded Factorio. They download mods from the portal, and after finishing the game they disable the mods and play with some other mods.
+
+![Factorio mod list with all Bob Angel's mods](https://raw.githubusercontent.com/LovelySanta/Alt-F4/angels-unit-tests/submissions/angels_unit_tests/mod_list.png)
+
+With these 4 scripts, everything is in place to launch factorio with a certain configuration of the latest mods and their mod settings. It was only a matter of telling factorio to launch over and over with different mod settings. Assuming an ingame mod (Angel's Dev - Unit Tests in this case) telling the testing infrastructure that the testing is over, it can shut down Factorio and launch the next configuration to test. At this point, the unit testing infrastructure could check if all configurations loaded without crashing. After loading the game, the unit testing mod could verify everything during the game, at runtime.
+
+## The last piece of the puzzle
+
+A week later, on the 8th of June 2021 I finished writing [the most basic unit test](https://github.com/Arch666Angel/mods/pull/634/commits/1435fbd2f1134b08307406e21fdd657f1cafa019) which did nothing more than reporting which mods are active ingame. After the unit test was finished logging the results, the [main script](https://github.com/Arch666Angel/mods/pull/634/commits/64ea68023f19c2505aa3cfbb7642f13cf7f70fbd) could launch the next configuration. A sample output of this single unit test can be seen below:
+
+```{r, class.output="scroll-100"}
```

Comment:
I think we decided to only show the the simple unit test, so it would be great if you could change it here. Maybe you could add an caption mentioning that this is only an example output.

Reply:
Resolved in https://github.com/AlternativeFFFF/Alt-F4/pull/551/commits/09e0500a041063c47fe716375139b83f823eb7e9.